### PR TITLE
ci(publish): fix getEnginesCommitHash

### PR DESCRIPTION
**Pull request by Jolg42**
Thanks @janpio for noticing that in https://github.com/prisma/prisma/actions/runs/6668971879/job/18125730860

### CODE REVIEW

File: scripts/ci/publish.ts [hunk 1]
```diff
@@ -657,9 +657,7 @@ Check them out at https://github.com/prisma/ecosystem-tests/actions?query=workfl
 async function getEnginesCommitHash(): Promise<string> {
   const prismaPath = path.resolve(process.cwd(), './packages/engines/package.json')
   const pkg = JSON.parse(await fs.promises.readFile(prismaPath, 'utf-8'))
-  // const engineVersion = pkg.prisma.version
```
Comment (janpio) [inline scripts/ci/publish.ts:660]: Any idea why this was commented out and not used in the first place?

Comment (janpio) [reply scripts/ci/publish.ts:660]: Ah, your suggestion actually goes into the `@prisma/engines-version` package.json and then gets `prisma.enginesVersion` there?

File: scripts/ci/publish.ts [hunk 2]
```diff
@@ -657,9 +657,7 @@ Check them out at https://github.com/prisma/ecosystem-tests/actions?query=workfl
 async function getEnginesCommitHash(): Promise<string> {
   const prismaPath = path.resolve(process.cwd(), './packages/engines/package.json')
   const pkg = JSON.parse(await fs.promises.readFile(prismaPath, 'utf-8'))
```
Comment (millsp) [inline scripts/ci/publish.ts:659]: If we use `import`, we can benefit from type-safety on the json properties

File: scripts/ci/publish.ts [hunk 3]
```diff
@@ -657,9 +657,7 @@ Check them out at https://github.com/prisma/ecosystem-tests/actions?query=workfl
...
-  // const engineVersion = pkg.prisma.version
-  const engineVersion = pkg.devDependencies['@prisma/engines-version']?.split('.').slice(-1)[0]
-
+  const engineVersion = pkg.devDependencies['@prisma/engines-version'].prisma.enginesVersion
```
Comment (millsp) [inline scripts/ci/publish.ts:660]: I don't think this property exists in `./packages/engines/package.json`. I'd suggest using a direct import from `import { prisma } from "@prisma/engines-version"` as suggested [here](https://github.com/prisma/prisma/pull/21653/files#r1374771640) or some similar approach.

File: scripts/ci/publish.ts (hunk 1 above)
Comment (Jolg42) [reply scripts/ci/publish.ts:660]: Yes, that's the best way to do this. We put the hash in the package.json

Comment (Jolg42) [reply scripts/ci/publish.ts:660]: The comment is about a different hash, not the one we are interested here

**Review by millsp — APPROVED** [review]

**Pull request merged**